src/dataDisplay/Identicon/index.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+import makeBlockie from 'ethereum-blockies-base64';
+import { RefObject, useEffect, useState } from 'react';
+
+type Props = {
+  address: string;
+  diameter: number;
+};
+
+type StyleProps = {
+  width: number;
+  height: number;
+};
+
+const getStyleFrom = (diameter: number): StyleProps => ({
+  width: diameter,
+  height: diameter
+});
+
+const generateBlockieIdenticon = (
+  address: string,
+  diameter: number
+): HTMLImageElement => {
+  const image = new window.Image();
+  image.src = makeBlockie(address);
+  image.height = diameter;
+  image.width = diameter;
+  image.style.borderRadius = `${diameter / 2}px`;
+
+  return image;
+};
+
+const Identicon = (props: Props) => {
+  const { address, diameter } = props;
+  const style: StyleProps = getStyleFrom(diameter);
+
+  const [identicon, setIdenticon] = useState<RefObject<HTMLImageElement> | null>(null);
+
+  useEffect(() => {
+    setIdenticon(React.createRef<HTMLImageElement>());
+  }, []);
+
+  useEffect(() => {
+    const image = generateBlockieIdenticon(address, diameter);
+
+    if (!identicon || !identicon.current) {
```

Comment:
you can simplify the condition in this way: `if(!identicon?.current)`

Also, you could move this block to the first sentence, so if it is not identicon attached, the image shouldn't be created.